Major improvements to addon system

npc-crow opened the pull request:
Added better catagorization, support for .json files inside of vpk/zip files (forgot to test vpk but i'm faiirly sure it'll work?) to identify the mod with, some other stuff probably i'm tired.

npc-crow committed:
Major improvements to addon system

Added better catagorization, support for .json files inside of vpk/zip files (forgot to test vpk but i'm faiirly sure it'll work?) to identify the mod with, some other stuff probably i'm tired.

cueki commented on app.py:
```diff
@@ -219,38 +221,55 @@ def on_addon_select(self):
-            if addon_groups[addon_type]:
-                for addon_name in sorted(addon_groups[addon_type]):
-                    item = QListWidgetItem(addon_name)
+            print(addon_type)
```
try not to leave any print statements (unless they are contextual)

cueki commented on app.py:
```diff
@@ -219,38 +221,55 @@ def on_addon_select(self):
+
 
         for addon in addons_dir.glob("*.zip"):
+#        for addon in self.addons_file_paths:
```
commented code

cueki commented on app.py:
```diff
@@ -331,22 +350,30 @@ def show_success(self, message):
-                "type": "Misc",
-                "description": "Information unavailable.",
-                "contents": []
+            print(f'Problem loading mod.json: {e}')
```
probably a better way to handle a missing mod.json but ill let it slide for now... 

FYI, the zip files for addons are on the repo, they should be updated with this pull request to reflect this change.

cueki commented on app.py:
```diff
@@ -219,38 +221,55 @@ def on_addon_select(self):
-            for addon_name in sorted(addon_groups["custom"]):
-                item = QListWidgetItem(addon_name)
-                self.addons_list.addItem(item)
+                print(addon_groups)
```
same deal here, I don't want to flood console (unless its an error)

cueki reviewed (changes requested):
A couple small things, but mainly just make sure the current addons support, and are properly updated into this new mod.json format :)

EmilyEmbers commented:
All very fair critiques - like i said, rusty and did this at like 3 am. Will fix later.

EmilyEmbers commented:
Having some issues with some bugs, I'll keep you updated.

npc-crow committed:
Improvements for full release

Fixed:
1. unknown display type console error
2. fixed issue with files in root of vpk not loading (should work across all vpk's)
3. improved app.py/load_addon_info for clarity and best practices
4. adjusted all addons to work with new mod.json system
5. removed unnecessary comments

I have not bothered to figure out why vpk_handler.py/list_files has issues with root paths and turns them into dots, but this at least functions.

EmilyEmbers commented:
I just pushed a commit that should fix all the problems you've had with the pull request. Let me know if there's any other issues.

cueki reviewed (approved):
Looks good, ill test more tomorrow to make sure. I can always update vpk_handler to handle paths better, I haven't looked at that code in a month or two.
